apps/frontend/src/libs/hooks/use-pagination/use-pagination.ts

```diff
@@ -0,0 +1,79 @@
+import {
+	useCallback,
+	useEffect,
+	useSearchParams,
+	useState,
+} from "~/libs/hooks/hooks.js";
+
+import { FIRST_PAGE } from "./libs/constants/constants.js";
+import { QueryParameter } from "./libs/enums/enums.js";
+import {
+	calculateTotalPages,
+	getInitialPage,
+	getInitialPageSize,
+} from "./libs/helpers/helpers.js";
+
+type Parameters = {
+	totalItems: number;
+};
+
+type ReturnType = {
+	onPageChange: (page: number) => void;
+	onPageSizeChange: (pageSize: number) => void;
+	page: number;
+	pageSize: number;
+};
+
+const usePagination = ({ totalItems }: Parameters): ReturnType => {
+	const [searchParameters, setSearchParameters] = useSearchParams();
+
+	const pageQueryParameter = searchParameters.get(QueryParameter.page);
+	const pageSizeQueryParameter = searchParameters.get(QueryParameter.pageSize);
+
+	const initialPageSize = getInitialPageSize(pageSizeQueryParameter);
+	const initialTotalPages = calculateTotalPages(initialPageSize, totalItems);
+	const initialPage = getInitialPage(pageQueryParameter, initialTotalPages);
```

Comment:
We don't want this to be calculated on each re-render. There as an ability to pass function to useState to compute initial state, can't we do it this way?

Reply:
Sure. Don't know about this ability ^__^